swift-sdk/Internal/EmptyInAppManager.swift

```diff
@@ -0,0 +1,53 @@
+//
+//
+//  Created by Tapash Majumder on 5/31/19.
+//  Copyright © 2019 Iterable. All rights reserved.
+//
+
+import Foundation
+
+class EmptyInAppManager : IterableInAppManagerProtocolInternal {
+    func start() {
+    }
+    
+    func createInboxMessageViewController(for message: IterableInAppMessage) -> UIViewController? {
+        ITBError("Can't create VC")
+        return nil
+    }
+    
+    func getMessages() -> [IterableInAppMessage] {
+        return []
+    }
+    
+    func getInboxMessages() -> [IterableInAppMessage] {
+        return []
+    }
+    
+    func show(message: IterableInAppMessage) {
+    }
+    
+    func show(message: IterableInAppMessage, consume: Bool, callback: ITBURLCallback?) {
+    }
+    
+    func remove(message: IterableInAppMessage) {
+    }
+    
+    func set(read: Bool, forMessage message: IterableInAppMessage) {
+    }
+    
+    func getUnreadInboxMessagesCount() -> Int {
+        return 0
+    }
+    
+    func onInAppSyncNeeded() -> Future<Bool, Error> {
```

Comment:
What is this `Future` used for?

Reply:
Signifies completion of sync. It will be mostly ignored.

Reply:
Is it only ever used in tests? I'm not a big fan of complicating code to simplify tests, but it's up to you.

Reply:
This is needed for more than tests. If a sync request arrives before another finishes, this chains the requests.